### PR TITLE
Task validation

### PR DESCRIPTION
**Pull request by Melvin-Tan**
#75 #119 
READY FOR REVIEW

The exceptions part is quite trivial, just put a bigass try catch in the controller and catch all illegalargumentexceptions.

The main part of this PR is really about making our add/update result message more meaningful than just "Task added" and "Task updated"

### CODE REVIEW

**Comment by louislai** [comment]
wha_ ab_ valida_ion for duplica_e _asks?

**Comment by Melvin-Tan** [comment]
will handle. wait for update before u wake up lol

File: src/main/java/seedu/toluist/controller/UpdateTaskController.java [hunk 1]
```diff
@@ -93,45 +101,71 @@ private CommandResult update(Task task, String description,
             boolean isFloating, String taskPriority, Set<Tag> tags,
             String recurringFrequency, LocalDateTime recurringUntilEndDate, boolean isStopRecurring) {
         if (!isValidTaskType(eventStartDateTime, eventEndDateTime, taskDeadline, isFloating)) {
```
Comment (louislai) [inline src/main/java/seedu/toluist/controller/UpdateTaskController.java:103]: what abt checking for duplicate after updating

Comment (Melvin-Tan) [reply src/main/java/seedu/toluist/controller/UpdateTaskController.java:103]: Handled below

**Comment by louislai** [comment]
make sure _his is hndled as well. `add` command silen_ly failed

File: src/main/java/seedu/toluist/controller/UpdateTaskController.java [hunk 2]
```diff
@@ -93,45 +101,73 @@ private CommandResult update(Task task, String description,
...
+                taskCopy.setStartDateTime(null);
+                taskCopy.setEndDateTime(taskDeadline);
+            } else {
+                if (eventStartDateTime != null) {
+                    taskCopy.setStartDateTime(eventStartDateTime);
+                }
+                if (eventEndDateTime != null) {
+                    taskCopy.setEndDateTime(eventEndDateTime);
+                }
             }
```
Comment (louislai) [inline src/main/java/seedu/toluist/controller/UpdateTaskController.java:133]: what is this monster metho o.O

File: src/main/java/seedu/toluist/controller/AddTaskController.java [hunk 3]
```diff
@@ -78,52 +79,68 @@ private CommandResult add(TodoList todoList, String description,
...
-            task.setTaskPriority(taskPriority);
-        }
-        if (StringUtil.isPresent(recurringFrequency)) {
-            if (recurringUntilEndDate == null) {
-                task.setRecurring(recurringFrequency);
+        try {
+            // validates that the dates input belongs to only one type of task, or exception is thrown
+            validateTaskDatesInput(eventStartDateTime, eventEndDateTime, taskDeadline);
+            Task task;
+            if (eventStartDateTime != null && eventEndDateTime != null) {
```
Comment (louislai) [inline src/main/java/seedu/toluist/controller/AddTaskController.java:89]: actually isnt this quite similar to the validation logic in update task?

Comment (Melvin-Tan) [reply src/main/java/seedu/toluist/controller/AddTaskController.java:89]: They are not exact. Update can update either from/ or to/
Add must have both arguments.

Comment (Melvin-Tan) [reply src/main/java/seedu/toluist/controller/AddTaskController.java:89]: And add doesnt have a floating/ argument, while update has.

**Review by louislai — APPROVED** [review]

**Pull request merged**